### DISCUSSION TITLE
Why does docgen ignore explicit typing?

### DISCUSSION THREAD

**Discussion by kylemh** [storybookjs/storybook#18644] [Help]
If I have some component:
```tsx
const BaseButton = (props: BaseButtonProps) => <button {...props} />;

const StyledButton = styled(BaseButton)``;

export type ButtonProps = {}; // unimportant
export const Button = (props: ButtonProps) =>  <StyledButton {...props} />;
```

and I import `Button` into a Storybook file like so:

```tsx
const meta: Meta<ButtonProps> = {
  component: Button,
  title: '/Button',
};
```

My Storybook instance will populate the table with props from `BaseButtonProps`. Not `ButtonProps`.It can lead to incorrect documentation if - for example - I Omit or Pick props from the Base* Props to define a different set of Props for the component being extended.

Why does it do this? Why doesn't docgen instead simply render tables based upon the type that the item is _on_ import?

**Comment by shilman** [comment]
Storybook use `react-docgen-typescript` to infer argTypes. I believe this strange behavior is an artifact of that library.

**Reply by kylemh** [reply]
🤦🏼 i was flustered with my experience, but I realize now I should've obviously posted in that repository. I'll move it there.

https://github.com/styleguidist/react-docgen-typescript/issues/440

**Reply by shilman** [reply]
We should have a better handle on all this stuff in Storybook. Coming in 7.x most likely!
